docs/uml-class-diagrams/middleware02/production/MiddlewareCommLinkManagerImplementation/MiddlewareCommLinkManagerImplementation.puml

```diff
@@ -4,7 +4,11 @@ package edu.oakland.production.middleware02 {
     class MiddlewareCommLinkManagerImplementation {
         __
         .. Use Case 1 ..
-
+        + TrackData parseRfid(int rfid)
+        float calculateSpeed()
+        float calculateDirection()
```

Comment:
Thoughts on breaking this into it's own helper class, like `TrackDataCalculator`? That would allow you to test the math methods without the rest of the overhead, and more evenly divide up development work on the team.

If so, let me know and I can get the file structure set up for that. If you wanted to go that direction, these could be static, so that choice will not impact your constructor or private variables.

If not, then make sure that these get marked as private!